src/Umbraco.Core/Persistence/SqlSyntax/SqlCeSyntaxProvider.cs

```diff
@@ -132,6 +132,16 @@ public override IEnumerable<ColumnInfo> GetColumnsInSchema(IDatabase db)
                     item => new Tuple<string, string, string, bool>(item.TABLE_NAME, item.INDEX_NAME, item.COLUMN_NAME, item.UNIQUE));
         }
 
+        /// <inheritdoc />
+        public override string GetDefaultConstraint(IDatabase db, string tableName, string columnName)
+        {
+            // cannot return a true default constraint name (does not exist on SqlCe)
+            // but we won't really need it anyways - just check whether there is a constraint
+            var hasDefault = db.Fetch<bool>(@"select column_hasdefault from information_schema.columns
+where table_name=@0 and column_name=@1", tableName, columnName).FirstOrDefault();
+            return hasDefault ? "XXXXX" : string.Empty;
```

Comment:
I'm not sure I understand this!
So if `"XXXXX"` does not matter? why can't we use `string.Empty` in both cases?

Reply:
there is a reason, but if you ask, it means the whole thing is not clear enough - looking into it

Reply:
this was using the constraint name both as a way to indicate "whether there is a constraint" and the actual constraint name (which does not exist on SqlCe) - have refactored to avoid the confusion and separate concerns